### PR TITLE
fix: unify HTTPS detection with cacti_is_https() helper

### PR DESCRIPTION
**Pull request by somethingwithproof**
## Summary
- Add `cacti_is_https()` helper to `lib/functions.php` that centralizes HTTPS detection
- Replace 8 inline `$_SERVER['HTTPS']` checks across 4 files with the new helper
- Handles edge cases: 'off' (case-insensitive), '0', empty string, and unset

## Files changed
- `lib/functions.php` -- new `cacti_is_https()` function, replace 4 inline checks
- `include/global.php` -- replace 2 inline checks (force_https redirect, cookie_secure)
- `graph_xport.php` -- replace 1 inline check
- `auth_login.php` -- replace 1 inline check
- `tests/Unit/CactiIsHttpsTest.php` -- 10 Pest tests covering all edge cases

## Test plan
- [ ] Verify HTTPS redirect still works with force_https enabled
- [ ] Verify session cookies set secure flag over HTTPS
- [ ] Verify CSV export Pragma header set over HTTPS

Signed-off-by: Thomas Vincent <thomasvincent@gmail.com>

### CODE REVIEW

File: lib/functions.php [hunk 1]
```diff
@@ -8317,6 +8317,18 @@ function get_debug_prefix() : string {
...
+/**
+ * Determine if the current request was made over HTTPS.
+ *
+ * Handles edge cases where $_SERVER['HTTPS'] may be empty, 'off',
+ * or mixed-case. Centralizes detection to avoid inconsistent checks.
+ *
+ * @return bool
+ */
+function cacti_is_https() : bool {
+	return isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== '' && strtolower($_SERVER['HTTPS']) !== 'off';
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply lib/functions.php]: Fixed. Added explicit handling for `'0'` as non-HTTPS and removed the unrelated `api_aggregate.php` changes. Branch rebased clean onto upstream develop.

File: tests/Unit/CactiIsHttpsTest.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+test('returns true when HTTPS is "1"', function () {
+	$_SERVER['HTTPS'] = '1';
+	expect(cacti_is_https())->toBeTrue();
+});
+
+test('returns true when HTTPS is any non-off truthy string', function () {
+	$_SERVER['HTTPS'] = 'anything';
+	expect(cacti_is_https())->toBeTrue();
+});
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply tests/Unit/CactiIsHttpsTest.php:53]: Fixed. Added test for `$_SERVER['HTTPS'] = '0'` (now 10 tests total).

File: lib/api_aggregate.php [hunk 3]
```diff
@@ -672,8 +668,6 @@ function aggregate_validate_graph_items(array $posted, array &$graph_items) : vo
 
 			if (isset($graph_items[$graph_templates_item_id])) {
 				$graph_items[$graph_templates_item_id]['item_total'] = $val;
-			} else {
-				cacti_log('Something fubar in agg_total');
 			}
 		}
 	}
```
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply lib/api_aggregate.php:679]: Fixed. Removed `api_aggregate.php` from this PR. Branch rebased onto upstream develop with only HTTPS-related changes.

[the remark this answers is not in the document]
Comment (somethingwithproof) [reply lib/api_aggregate.php:679]: Removed.

File: lib/functions.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply lib/functions.php]: Intentional. The helper normalizes all checks to a single codepath.

File: tests/Unit/CactiIsHttpsTest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (somethingwithproof) [reply tests/Unit/CactiIsHttpsTest.php:53]: Already covered at line 67.

**Comment by somethingwithproof** [comment]
Scope cleanup update: I removed the unrelated `lib/api_aggregate.php` changes in `b2ab5466` (`chore: drop unrelated lib/api_aggregate changes`) to keep this PR focused and avoid duplicate problem-solving across PRs.

**Review by TheWitness — DISMISSED** [review]

**Review by TheWitness — APPROVED** [review]

**Comment by somethingwithproof** [comment]
Reviewed existing Copilot/TheWitness feedback threads on this PR; current branch already contains the requested fixes (including HTTPS value 0 handling and prior scope cleanup), so no additional code change was needed in this pass.

**Review by TheWitness — APPROVED** [review]

**Review by TheWitness — DISMISSED** [review]

**Review by TheWitness — APPROVED** [review]
